### ISSUE TITLE
Change recommended React Router usage in Dropdowns

### ISSUE THREAD

**Issue by SpyTec** [patternfly/patternfly-react#3720]
Right now, in Dropdown docs we recommend using
```jsx
<DropdownItem key="link">
  <Link to={'/'}>Link</Link>
</DropdownItem>
```

Since this by default will create nested `<a>` we will get an error. On top of that there will also be applied global link styling.

One way we could fix this is by using hooks.
```jsx
const history = useHistory();
//...

<DropdownItem key="link" onClick={() => history.push("/")}>
  Link
</DropdownItem>
```

And for class-based usage we can recommend usage from this [StackOverflow answer](https://stackoverflow.com/a/42679052). That way we can use it the same way with `onClick` as shown above.

**Comment by kmcfaul** [comment]
Adding the property `component="div"` will also prevent the nesting anchor issue. Our example will have to be updated to include that, at the very least.

**Comment by SpyTec** [comment]
Then we still have the issue described on #3641, where the link will have link styling applied as well as only registering when you click the actual text instead of somewhere in the dropdown item

**Comment by SpyTec** [comment]
Fixed in #3889
